src/VPUXCompilerL0/CMakeLists.txt

```diff
@@ -75,10 +75,23 @@ endif()
 target_include_directories(${FUNCTIONAL_TARGET}
     PUBLIC
         "${CMAKE_SOURCE_DIR}/src/core/include"
-    PRIVATE
-        $<TARGET_PROPERTY:IE::nlohmann_json,INTERFACE_INCLUDE_DIRECTORIES>
 )
 
+if(TARGET IE::nlohmann_json)
+    target_include_directories(${FUNCTIONAL_TARGET}
+        PRIVATE
+        $<TARGET_PROPERTY:IE::nlohmann_json,INTERFACE_INCLUDE_DIRECTORIES>)
+else()
+    find_package(nlohmann_json 3.9.0 QUIET)
+    if(nlohmann_json_FOUND)
+        target_include_directories(${FUNCTIONAL_TARGET}
+            PRIVATE
+            $<TARGET_PROPERTY:nlohmann_json::nlohmann_json,INTERFACE_INCLUDE_DIRECTORIES>)
+    else()
+        message(FATAL_ERROR "Failed to find system nlohmann_json in OpenVINO Developer Package and system libraries")
+    endif()
+endif()
+
```

Comment:
14 lines instead of 14 lines :) 
anyway, nice solution. However, let's make it as it is in the internal master branch

Reply:
It's a bit less code just because you don't have to separate one `target_include_directories` call from another with conditional dependency

I thought it is a good idea to refactor it here just because this cherry-pick will be unrefactored forever before a new snapshot (probably not even the next one) is released

Reply:
Anyway, we can return to it later/wait until this workaround is not even needed

Reply:
This WA will be not required since migration to yaml-cpp